speech/speechrest.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python
+# Copyright 2016 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# See the README for more information about how to run this sample.
+# Make sure that the Speech API is enabled first:
+# developers.google.com/apis/api/speech.googleapis.com/overview
+# You must be whitelisted to do this.
+#
+# The example uses a service account for OAuth2 authentication.
+# You may obtain service account credentials at:
+# https://console.developers.google.com . Navigate to:
+# API Manager > Credentials > Create credentials >
+# Service account key > New service account.
+# Create a new service account, and download the json credentials file.
+# Then set the environment variable 'GOOGLE_APPLICATION_CREDENTIALS' to the
+# full path of that file.
+
+
+"""Google Cloud Speech API sample application using the REST API for batch
+processing."""
+
+# [START import_libraries]
+import argparse
+import base64
+import json
+
+from googleapiclient import discovery
+
+import httplib2
+
+from oauth2client.client import GoogleCredentials
+# [END import_libraries]
+
+# Path to local discovery file
+# [START discovery_doc]
+API_DISCOVERY_FILE = 'speech-discovery_google_rest_v1.json'
+# [END discovery_doc]
+
+
+# Authenticate using a Service Account.
+# Application default credentials provided by env variable
+# GOOGLE_APPLICATION_CREDENTIALS
+def get_speech_service():
+    # [START authenticating]
+    credentials = GoogleCredentials.get_application_default().create_scoped(
+        ['https://www.googleapis.com/auth/xapi.zoo'])
+    with open(API_DISCOVERY_FILE, 'r') as f:
+        doc = f.read()
+    return discovery.build_from_document(
+        doc, credentials=credentials, http=httplib2.Http())
+    # [END authenticating]
+
+
+# The main program accepts a single argument, which is the name of the audio
+# raw file.
+def main(speech_file):
+    # [START construct_request]
+    service = get_speech_service()
+    with open(speech_file, 'rb') as speech:
+        # Base64 encode the binary audio file for inclusion in the JSON
+        # request.
+        speech_content = base64.b64encode(speech.read())
+        service_request = service.speech().recognize(
```

Comment:
Since you're done with the file, there's no need for this line and below to be in the context manager.